src/test.cc

```diff
@@ -75,10 +75,12 @@ char* mkdtemp(char* name_template) {
 
 string GetSystemTempDir() {
 #ifdef _WIN32
-  char buf[1024];
-  if (!GetTempPath(sizeof(buf), buf))
+  TCHAR buf[MAX_PATH];
+  // GetTempPath returns zero if no path is found
+  if(GetTempPath(MAX_PATH, buf) == 0)
     return "";
```

Comment:
Returning an empty string seems like the wrong solution.

I'd rather see an exception get thrown, or the function accept something like `std::error_code` as a reference (e.g. how `std::filesystem` does it)

Reply:
Just as a reminder: Ninja doesn't use exceptions.

Reply:
Then modify the compiler configuration to tell the compiler that.

At the least, the CMake build does not specify this. I have not checked the python build script.

Reply:
Pretty sure RTTI should be disabled as well, if you're going to disable things. I don't believe RTTI is used anywhere in Ninja.

Reply:
The empty return is the same as before, so it should not change any behavoiur.

Reply:
See https://github.com/ninja-build/ninja/issues/2013 for disabling exceptions and RTTI at compile time.